placehero/src/html_content.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#[derive(Debug)]
+enum TagCloseBehaviour {
+    /// Nothing needs to happen when the span is closed (i.e. we didn't do anything for it?)
+    None,
+    /// A `span.invisible` is ending (and so `emit` should be toggled back on)
+    Hidden,
+    /// An ellipisis should be output at the end of this item.
+    Ellipsis,
+    /// A paragraph.
+    Paragraph,
+}
+
+/// Convert sanitised HTML into a displayable string.
+///
+/// See <https://docs.joinmastodon.org/spec/activitypub/#sanitization> for the tags we have to support.
+///
+/// Note:
+/// 1) We convert HTML entities to their regular value (hopefully?)
+/// 2) We only handle the `p`, `br`, `span.invisible`, `span.ellipsis` cases
+/// 3) We don't handle `microformat` at all.
+// TODO: We know this code is not great (and probably way too imperative!)
+// We're deferring refactoring this until we want to handle more attributes.
+pub(crate) fn handle_content_html(content: &str) -> String {
```

Comment:
`handle_content_html` -> `textify_mastodon_html` or `convert_mastodon_html_to_text`?

Reply:
I've renamed to `status_html_to_plaintext`

Reply:
Sounds good.